oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleDBRecord.java

```diff
@@ -50,31 +61,208 @@ protected SchemaReader getSchemaReader() {
+   * Retrieves bytes representation of 'oracle.sql.BFILE' via 'oracle.sql.BFILE#getBytes'. Note, that this method
+   * retrieves only locator (link) to an external binary file (file stored outside of the database) and not content of
+   * the file. <a href="https://docs.oracle.com/cd/B19306_01/java.102/b14355/oralob.htm#BABJJEIC">Files are created
+   * only externally.</a>
+   * @param bfile instance of 'oracle.sql.BFILE'.
+   * @return bytes representation of 'oracle.sql.BFILE' obtained via 'oracle.sql.BFILE#getBytes'.
+   */
+  private byte[] getBfileBytes(ResultSet resultSet, Object bfile) {
+    if (bfile == null) {
+      return null;
+    }
+    try {
+      ClassLoader classLoader = resultSet.getClass().getClassLoader();
+      Class<?> bfileClass = classLoader.loadClass("oracle.sql.BFILE");
+      return (byte[]) bfileClass.getMethod("getBytes").invoke(bfile);
+    } catch (ClassNotFoundException e) {
+      throw new InvalidStageException("Unable to load 'oracle.sql.BFILE'.", e);
+    } catch (InvocationTargetException | NoSuchMethodException | IllegalAccessException e) {
+      throw new InvalidStageException("Error while invoking 'oracle.sql.BFILE#getBytes()'.", e);
+    }
+  }
+
+  private void handleOracleSpecificType(ResultSet resultSet, StructuredRecord.Builder recordBuilder, Schema.Field field,
+                                        int columnIndex, int sqlType, int precision, int scale)
+    throws SQLException {
     switch (sqlType) {
       case OracleSchemaReader.INTERVAL_YM:
       case OracleSchemaReader.INTERVAL_DS:
+      case OracleSchemaReader.LONG:
       case Types.NCLOB:
         recordBuilder.set(field.getName(), resultSet.getString(columnIndex));
         break;
-      case OracleSchemaReader.TIMESTAMP_LTZ:
       case OracleSchemaReader.TIMESTAMP_TZ:
+        recordBuilder.set(field.getName(), resultSet.getString(columnIndex));
+        break;
+      case OracleSchemaReader.TIMESTAMP_LTZ:
         Instant instant = resultSet.getTimestamp(columnIndex).toInstant();
         recordBuilder.setTimestamp(field.getName(), instant.atZone(ZoneId.ofOffset("UTC", ZoneOffset.UTC)));
         break;
+      case OracleSchemaReader.BINARY_FLOAT:
+        recordBuilder.set(field.getName(), resultSet.getFloat(columnIndex));
+        break;
+      case OracleSchemaReader.BINARY_DOUBLE:
+        recordBuilder.set(field.getName(), resultSet.getDouble(columnIndex));
+        break;
+      case OracleSchemaReader.BFILE:
+        // Note, that ResultSet#getObject retrieves only locator (link) to an external binary file (file stored outside
+        // of the database) and not content of the file.
+        Object bfile = resultSet.getObject(columnIndex);
+        recordBuilder.set(field.getName(), getBfileBytes(resultSet, bfile));
+        break;
+      case OracleSchemaReader.LONG_RAW:
+        recordBuilder.set(field.getName(), resultSet.getBytes(columnIndex));
+        break;
+      case Types.DECIMAL:
+      case Types.NUMERIC:
```

Comment:
Yes,  they come back as NUMERIC.

'Get Schema' button returns decimal for those columns, the exported pipeline JSON file will contain the following entry:
```
"schema": "{\"type\":\"record\",\"name\":\"etlSchemaBody\",\"fields\":[{\"name\":\"INT_COL\",\"type\":[{\"type\":\"bytes\",\"logicalType\":\"decimal\",\"precision\":38},\"null\"]},{\"name\":\"INTEGER_COL\",\"type\":[{\"type\":\"bytes\",\"logicalType\":\"decimal\",\"precision\":38},\"null\"]},{\"name\":\"SMALL\",\"type\":[{\"type\":\"bytes\",\"logicalType\":\"decimal\",\"precision\":38},\"null\"]}]}"
```

We read them as:
https://github.com/data-integrations/database-plugins/pull/46/files#diff-aef8f5d3e726ab43b5650285f66bcc1fR219
```
      case Types.DECIMAL:
      case Types.NUMERIC:
        // This is the only way to differentiate FLOAT/REAL columns from other numeric columns, that based on NUMBER.
        // Since FLOAT is a subtype of the NUMBER data type, 'getColumnType' and 'getColumnTypeName' can not be used.
        if (Double.class.getTypeName().equals(resultSet.getMetaData().getColumnClassName(columnIndex))) {
          recordBuilder.set(field.getName(), resultSet.getDouble(columnIndex));
        } else {
          // It's required to pass 'scale' parameter since in the case of Oracle, scale of 'BigDecimal' depends on the
          // scale of actual value. For example for value '77.12' scale will be '2' even if sql scale is '6'
          BigDecimal decimal = resultSet.getBigDecimal(columnIndex, scale);
          recordBuilder.setDecimal(field.getName(), decimal);
        }
```

Currently, if precision/scale of the explicit input schema does not match sql precision/scale, the Source will fail. Should we do rounding in this case?